libs/transactions/src/lib/views/transaction-details/transaction-details.component.ts

```diff
@@ -0,0 +1,26 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, NavigationExtras, Router } from '@angular/router';
+
+@Component({
+  templateUrl: './transaction-details.component.html',
+  selector: 'bb-transaction-details',
+})
+export class TransactionDetailsComponent implements OnInit {
+  public transactionDetails;
+  private routerState: NavigationExtras | undefined;
+  constructor(public route: ActivatedRoute, private router: Router) {
+    this.routerState = this?.router?.getCurrentNavigation()?.extras.state;
```

Comment:
The best practice is to resolve the data in the `Router.resolve` and transfer to the component only ready data

Reply:
Added a resolver but for fetching the data from the API I have created a TODO task